build.py

```diff
@@ -1,160 +1,127 @@
 #! /usr/bin/env python3
 
 import argparse
-import os
 from pathlib import Path
-from string import Template
 import subprocess
 import sys
-
-TOP_DIR = Path(__file__).parent.resolve()
-PSY_DIR = TOP_DIR / "psy"
-SRC_DIR = TOP_DIR / "src"
-CFG_DIR = TOP_DIR / "config"
-STUBS_DIR = TOP_DIR / "psy" / "stubs"
-
-AGENT_CONFIG_HJSON_TEMPLATE = """
-{
-    app: {
-        agents: [
-            {
-                name: $agent_name
-                core: $agent_core
-            }
-        ]
-    }
-}
-"""
-
-CORUNNER_CONFIG_HJSON_TEMPLATE = """
-{
-    app: {
-        cores: [
-            {
-                id: ${corunner_id}
-                co_runner: {
-                    symbol: ${corunner_symbol}
-                    object: ${corunner_object}
-                }
-            }
-        ]
-    }
-}
-"""
-
-COMPILE_CONFIG_HJSON_TEMPLATE = """
-{
-    compile: {
-        cflags: [
-            "-g2",
-            "-Xdebug-dwarf2",
-            "-Xdebug-local-cie",
-            "-Xdialect-c99",
-            "-X230=1",
-            "-X43=0", # Do not insert eieio !
-            """ + f'''"-I", "{TOP_DIR}/include",''' + """
-        ]
-    }
-}
-"""
-
-PSYMODULE_CONFIG_HJSON_TEMPLATE = """
-{
-    psymodule: {
-        psy_pp_flags: [
-            "-I", "include",
-        ]
-        cflags: [
-            "-g2",
-            "-Xdebug-dwarf2",
-            "-Xdebug-local-cie",
-            "-Xdialect-c99",
-            "-X230=1",
-            "-X43=0", # Do not insert eieio !!
-            """ + f'''"-I", "{TOP_DIR}/include",''' + """
-        ]
-    }
-}
-"""
-
-class Help:
-    RTK_DIR = "Path to the ASTERIOS RTK"
-    CORUNNER_ID = "ID of the co-runner to enable; can be specified multiple times"
-    TASK = "Name of the nominal task to be compiled for execution"
-    PSYKO = "Path to the PsyC Compiler psyko"
-    BUILD_DIR = "Path to the build directory in which artifacts are produced"
-    CORE = "ID of the core on which the task will run; must not conflict with --corunner-id"
-    LOCAL_CORUNNERS = "If set, co-runners will be configured to only access local memories"
-    OUTPUT = "Path where the executable is to be generated"
-    PRODUCT = "Name of the ASTERIOS RTK Product"
+import scripts.templates
+from scripts.templates import *
+from scripts.scriptutil import load_db, load_json, dump_json, write_template, psyko
+from operator import itemgetter
+
+def cor_par(s):
+  pars = s.split(',')
+  pars[0] = int(pars[0])
+  assert pars[0] in CORES, \
+    f"The corunner id must be one of {CORES.join(', ')}"
+  l = len(pars)
+  if l > 2:
+    raise argparse.ArgumentTypeError("Corunners parameters must be of type <core>,<start address of read>")
```

Comment:
if l == 1, I assume that the only specified parameter is a core. I added a case if pars[0] == '' to avoid the consequences of some mistakes made by the user.

Reply:
I realize that, but the error message states that only one "parameter" is not allowed.
It could be written as:
```suggestion
    raise argparse.ArgumentTypeError("Corunners parameters must be of type <core>[,<start address of read>]")
```